core/consensus/sumeragi.cpp

```diff
@@ -227,13 +236,21 @@ namespace sumeragi {
         connection::iroha::Sumeragi::Verify::receive([](const std::string& from, ConsensusEvent& event) {
             logger::info("sumeragi") << "receive!";
             logger::info("sumeragi") << "received message! sig:[" << event.eventsignatures_size() << "]";
-        
-            // send processTransaction(event) as a task to processing pool
-            // this returns std::future<void> object
-            // (std::future).get() method locks processing until result of processTransaction will be available
-            // but processTransaction returns void, so we don't have to call it and wait
-            std::function<void()> &&task = std::bind(processTransaction, event); 
-            pool.process(std::move(task)); 
+
+            logger::info("sumeragi") << "received message! status:[" << event.status() << "]";
+            if(event.status() == "commited") {
+                if(txCache.find(detail::hash(event.transaction())) == txCache.end()) {
+                    executor::execute(event.transaction());
+                    txCache[detail::hash(event.transaction())] = "commited";
+                }
+            }else{
+                // send processTransaction(event) as a task to processing pool
+                // this returns std::future<void> object
+                // (std::future).get() method locks processing until result of processTransaction will be available
+                // but processTransaction returns void, so we don't have to call it and wait
+                std::function<void()> &&task = std::bind(processTransaction, event);
+                pool.process(std::move(task));
```

Comment:
Sorry for the mistake, it seems a named rvalue referrence is a lvalue. Thanks, guys.